### PR TITLE
Fixed canceled order item statuses

### PR DESCRIPTION
**Pull request by SebaKas21**

### CODE REVIEW

**Review by dawiddworak88 — APPROVED** [review]
Looks good 👍

File: be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs [hunk 1]
```diff
@@ -864,20 +864,23 @@ private async Task MapStatusesToOrderStatusId(Guid? orderId)
...
-                            OrderItemStatusId = lastOrderItemStatusChange.OrderItemStatusId
+                            OrderItemStatusId = lastOrderItemStatusChange.OrderItemStatusId,
+                            CreatedDate = lastOrderItemStatusChange.CreatedDate
                         });
                     }
                 }
 
                 bool isSameStatus = lastOrderItemStatusChanges.DistinctBy(x => x.OrderItemStatusId).Count() == 1;
 
+                var lastOrderItemStatus = lastOrderItemStatusChanges.FirstOrDefault();
```
Comment (dawiddworak88) [inline be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs:874]: **suggestion:** let's add a null check for lastOrderItemStatus . If it's null there will be an exception on line 879-880.

**Review by dawiddworak88 — CHANGES_REQUESTED** [review]

File: be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs [hunk 2]
```diff
@@ -864,20 +864,23 @@ private async Task MapStatusesToOrderStatusId(Guid? orderId)
                         lastOrderItemStatusChanges.Add(new OrderItemStatusServiceModel
                         {
                             OrderItemStateId = lastOrderItemStatusChange.OrderItemStateId,
-                            OrderItemStatusId = lastOrderItemStatusChange.OrderItemStatusId
+                            OrderItemStatusId = lastOrderItemStatusChange.OrderItemStatusId,
+                            CreatedDate = lastOrderItemStatusChange.CreatedDate
```
Comment (dawiddworak88) [inline be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs]: **question:** wouldn't CreatedDate = DateTime.UtcNow be enough?

**Review by dawiddworak88 — CHANGES_REQUESTED** [review]

File: be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs (hunk 2 above)
Comment (SebaKas21) [reply be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs]: it was for testing, i removed

**Review by dawiddworak88 — APPROVED** [review]
👍

**Pull request merged**